cmake/tests.cmake

```diff
@@ -48,3 +48,51 @@ int main(int argc, char* argv[]) {\n\
   target_link_libraries(GTest_Main PUBLIC GTest::GTest)
   add_library(GTest::Main ALIAS GTest_Main)
 endif()
+
+if(NOT TARGET GTest::QtCoreMain AND TARGET Qt5::Core)
```

Comment:
Not sure I get why "test_qtcore_main.cpp" and "test_qtgui_main.cpp" aren't part of the repo, but created by the makefile. If we stick with this, shouldn't they at least be part of .gitignore so they are not accidentally added to the repo, in which case later changes to the makefile will not have any effect?

Reply:
They are created in the build-folder which is part of .gitignore in its entirety.

Reply:
Ah ok, that makes sense, thanks for clearing that up.